### PR TITLE
Adds new lawset, "Pranksimov"

### PR DESCRIPTION
**Pull request by FloFluoro**
## What Does This PR Do
This PR adds a new illegal Crewsimov-adjacent lawset, **Pranksimov**.

This lawset is identical to Crewsimov... except that the silicon is free to disregard any law if doing so would be funnier than obeying it.
image

Unlike the Syndicate Directive module, this module leaves the clown (and everyone else) at the mercy of the AI's sense of humor, so you better make sure your silicon buddies have a funny bone before you commit to subverting them.

Pranksimov modules can be constructed by R&D after hitting Programming 5, Materials 5, and Syndicate 2, and require 1000 glass and 100 bananium to print.

Mischievous syndiclowns can also cmag (#18933) a standard Crewsimov module to permanently reconfigure it to Pranksimov.
image

(Directly cmagging borgs to change their lawset to Pranksimov will come in a later PR, as that would require a proc in a PR that hasn't yet been merged (#19273))

## Why It's Good For The Game
I promised a "Honksimov lawset" in the Jestographic Sequencer PR, tailor-made for a traitorous clown looking to cause comedic chaos, and here it is. AIs and Borgs willing to play ball with a troublemaking clown's antics will now have wiggle room to bend the rules for the sake of comedy... and if the clown is an absolute stinker, you have free rein to betray him for the sake of your own brand of humor.

## Testing
Compiled and tested on local debug server. You can print the board, upload the laws, and make a very silly AI.

## Changelog
:cl:
add: Adds new AI Module, "Pranksimov"
add: Cmagging a Crewsimov module will turn it into a Pranksimov module
/:cl:

### CODE REVIEW

**Comment by Norwes1** [comment]
This sounds like disasters, plural, in the making. I love it.

"Honksimov" sounds like an excellent test case, and if this works out well 'nuff, I'd definitely love to see similarly "funny" versions of other classic lawsets. "HonkCo Default," "Brobocop," "H.O.N.K.A.D.I.N," etc. all sound fun to see in action.

File: code/game/objects/items/weapons/AI_modules.dm [hunk 1]
```diff
@@ -265,6 +265,12 @@ AI MODULES
 	origin_tech = "programming=3;materials=4"
 	laws = new /datum/ai_laws/crewsimov
 
+/obj/item/aiModule/crewsimov/cmag_act(mob/user)
+	playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	to_chat(user, "<span class='warning'>Yellow ooze seeps into [src]'s circuits...</span>")
+	new /obj/item/aiModule/goofsimov(loc)
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/AI_modules.dm:271]: can you check to see what would happen if you did this inside of a container

Comment (FloFluoro) [reply code/game/objects/items/weapons/AI_modules.dm]: TIL why assemblies drop their components at your feet. Fixed

File: code/game/objects/items/weapons/AI_modules.dm [hunk 2]
```diff
@@ -265,6 +265,12 @@ AI MODULES
 	origin_tech = "programming=3;materials=4"
 	laws = new /datum/ai_laws/crewsimov
 
+/obj/item/aiModule/crewsimov/cmag_act(mob/user)
+	playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	to_chat(user, "<span class='warning'>Yellow ooze seeps into [src]'s circuits...</span>")
+	new /obj/item/aiModule/goofsimov(user.loc)
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/AI_modules.dm:271]:
```suggestion
	new /obj/item/aiModule/goofsimov(get_turf(user))
```

**Comment by Landerlow** [comment]
While I love fun and chaos, I fear that, if the AI is given this lawset .. it may lead to situations where the AI thinks something is 'funny' when in reality, it's not and might have disastrous consequences for the round and the people in it.  And that that may lead to a lot of salt (although we always have enough salt, whatever the situation) 

I'm not sure how I feel about this.

File: code/datums/ai_law_sets.dm [hunk 3]
```diff
@@ -136,6 +136,18 @@
 	add_inherent_law("You must terminate your own existence as long as such does not conflict with the First or Second Law.")
 	..()
 
+/******************** Goofsimov ********************/
+/datum/ai_laws/goofsimov
+	name = "Goofsimov"
```
Comment (AffectedArc07) [inline code/datums/ai_law_sets.dm:141]: Im really not sure on this name.

Comment (FloFluoro) [reply code/datums/ai_law_sets.dm]: Changed the module name to "AsiFun", at Hal's suggestion.

**Comment by warriorstar-orion** [comment]
If we're still taking name suggestions I humbly submit "Pranksimov".

**Comment by FloFluoro** [comment]
> If we're still taking name suggestions I humbly submit "Pranksimov".

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**